### DISCUSSION TITLE
No cop for `expect(foo.empty?).to be(true)` or `expect(foo.include?(whatever)).to be(true)`

### DISCUSSION THREAD

**Discussion by Earlopain** [rubocop-hq/rubocop-rspec#1925] [Q&A]
Hi there, am I missing something or are there just no cops that check for the above code at the moment?

I would expect these corrections:
```rb
expect(foo.empty?).to be(true)
expect(foo).to eq('')
# =>
expect(foo).to be_empty
```

```rb
expect(foo.include?(whatever)).to be(true)
# =>
expect(foo).to inlcude(whatever)
```

The reason for that is that on failure the message is much more explicit instead of just saying that true/false != false/true. There are some cops I would expect to take here like `RSpec/PredicateMatcher` or `RSpec/BeEmpty` but that's not happening.

If this is missing at the moment I'd love to implement this myself but will need some guideance to which cops this logic would belong to.

**Comment by pirj** [comment]
How about setting `strict: false` on https://docs.rubocop.org/rubocop-rspec/cops_rspec.html#strict-false-enforcedstyle-inflected?

Related issue https://github.com/rubocop/rubocop-rspec/issues/919

[BeEmpty](https://github.com/rubocop/rubocop-rspec/issues/919) is about empty arrays, not even Hashes.
A PR with new cops to cover {} and empty string with `eq`/`be` is welcome.

**Reply by Earlopain** [reply]
Yeah, I tried changing the settings of that cop but I'm not getting any new offenses with that.

Nice to know I at least didn't overlook something for the empty string case. I'll try to make a PR for that, thanks.
